plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -38,6 +39,7 @@
     private final CatalogName catalogName;
     private final String trinoVersion;
     private final URI serverUri;
+    private final Optional<String> warehouse;
```

Comment:
`defaultWarehouseDir` ?

Reply:
what is warehouse identifier?

are there any docs for that that we should reference from our docs?

Reply:
An identifier could be anything that the REST Server would use to map to the warehouse being addressed.  This could be as simple as something like `prod` which the server then would map to whatever the canonical location is for prod.  For the docs, we could just reference the spec, but there's no strict requirement on format or how a server uses the value.

A good example of why this is useful is if you have something like the JDBCCatalog where you have to explicitly define the location in the config, but if something like Flink is also using the same catalog backend, but unintentionally points to a different location, you have a single catalog with data that's split across multiple locations.  This can even result in individual tables having data end up split across multiple locations depending on writes/overwrites.